x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1071,6 +1069,35 @@ static void validateForFips(Settings settings) {
         }
     }
 
+    void possiblyValidateImplicitSecurityBehaviorOnUpdate(Settings settings, NodeMetadata nodeMetadata) {
+        if (null == nodeMetadata) {
+            // No Node metadata for given node
+            return;
+        }
+        final XPackLicenseState licenseState = getLicenseState();
+        if (licenseState.getLicenseExpiryDate() == Long.MAX_VALUE) {
```

Comment:
Is this condition ever false when running properly? We're running this from `createComponents`, but I think `licenseState` represents the cluster state that's exposed by the `ClusterService` so the state isn't even initialised when this runs, let alone derived from disk. Sorry if I'm missing something obvious here...

Reply:
It's probably me missing something obvious. Let me circle back when I revisit this, but if what you say is true we can never know the license state that early in order to make decisions based on it. Up until recently (#72300) we where creating components here that depended on the license state ( i.e the APIKeyService ) to determine if they were enabled.